modules/Essentiality.py

```diff
@@ -0,0 +1,370 @@
+    GENE_EXPRESSION = "OmicsExpressionProteinCodingGenesTPMLogp1.csv"
+    MUTATION_HOTSPOTS = "OmicsSomaticMutationsMatrixHotspot.csv"
+    MUTATION_DAMAGING = "OmicsSomaticMutationsMatrixDamaging.csv"
+
+    def __init__(
+        self: DepMapEssentiality,
+        spark: SparkSession,
+        input_folder: str,
+        tissue_mapping_url: str,
+        keep_only_essentials: bool = True,
+    ) -> None:
+        self.spark = spark
+
+        # Reading model data:
+        models_df = self._prepare_models(f"{input_folder}/{self.MODELS}")
+
+        # Get a list of essential genes:
+        essential_genes_df = self._prepare_essential_genes(
+            f"{input_folder}/{self.ESSENTIAL_GENE_LIST}"
+        )
+
+        # Get a list of effect for each gene/cell line pair:
+        gene_effect_df = self._read_and_melt(
+            f"{input_folder}/{self.GENE_EFFECTS}", "geneEffect"
+        )
+
+        # Read and process expression data:
+        expression_df = self._read_and_melt(
+            f"{input_folder}/{self.GENE_EXPRESSION}", "expression"
+        )
+
+        # Read and process hotspot mutation:
+        hotspots = self._read_and_melt(
+            f"{input_folder}/{self.MUTATION_HOTSPOTS}", "hotspotMutation"
+        ).filter(f.col("hotspotMutation") > 0)
+
+        # Read and process damaging mutation:
+        damaging_mutation = self._read_and_melt(
+            f"{input_folder}/{self.MUTATION_DAMAGING}", "damagingMutation"
+        ).filter(f.col("damagingMutation") > 0)
+
+        # Get tissue mapping:
+        tissue_mapping = self._get_tissue_mapping(tissue_mapping_url)
+
+        # Join all the data together:
+        self.essentials = (
+            # The melted effect table:
+            gene_effect_df
+            # Narrowing down the gene effect table to genes that are in the essential gene list:
+            .join(essential_genes_df, on="targetSymbol", how="left")
+            .repartition("depmapId")
+            # Joining with model information:
+            .join(models_df, on="depmapId", how="left")
+            # Joining expression data:
+            .join(expression_df, on=["targetSymbol", "depmapId"], how="left")
+            # Joining damaging mutation data:
+            .join(damaging_mutation, on=["targetSymbol", "depmapId"], how="left")
+            # Joining hotspot mutation data:
+            .join(hotspots, on=["targetSymbol", "depmapId"], how="left")
+            # Joining tissue mapping:
+            .join(tissue_mapping, on="tissueFromSource", how="left")
+            # Format data:
+            .select(
+                "targetSymbol",
+                "depmapId",
+                # Cell-line fields:
+                "cellLineName",
+                f.col("modelId").alias("diseaseCellLineId"),
+                # Disease:
+                "diseaseFromSource",
+                # Tissue fields:
+                "tissueId",
+                f.when(f.col("tissueName").isNull(), "other")
```

Comment:
you can use `coalesce` for this type of operations

Reply:
Done.